src/drc/api/viewsets.py

```diff
@@ -118,16 +143,50 @@ class EnkelvoudigInformatieObjectViewSet(NotificationViewSetMixin,
     notifications_kanaal = KANAAL_DOCUMENTEN
     audit = AUDIT_DRC
 
+    @transaction.atomic
     def perform_destroy(self, instance):
-        if instance.objectinformatieobject_set.exists():
+        if instance.canonical.objectinformatieobject_set.exists():
             raise serializers.ValidationError({
                 api_settings.NON_FIELD_ERRORS_KEY: _(
                     "All relations to the document must be destroyed before destroying the document"
                 )},
                 code="pending-relations"
             )
 
-        super().perform_destroy(instance)
+        super().perform_destroy(instance.canonical)
+
+    @property
+    def filterset_class(self):
+        """
+        To support filtering by versie and registratieOp for detail view
+        """
+        if self.detail:
+            return EnkelvoudigInformatieObjectDetailFilter
+        return EnkelvoudigInformatieObjectListFilter
+
+    def get_serializer_class(self):
+        """
+        To validate that a lock id is sent only with PUT and PATCH operations
+        """
+        if self.action == 'update' or self.action == 'partial_update':
+            return EnkelvoudigInformatieObjectWithLockSerializer
+        return EnkelvoudigInformatieObjectSerializer
+
+    def get_queryset(self):
+        """
+        Haal de meest recente versie van elk `EnkelvoudigInformatieObject` op
+        """
+        qs = super().get_queryset()
+        return qs.order_by('canonical', '-versie').distinct('canonical')
```

Comment:
this repeats the declared `order_by` and `distinct`, oversight? or is there a reason?